lib/ansible/modules/crypto/openssl_dhparam.py

```diff
@@ -171,6 +196,70 @@ def check(self, module):
             return False
         return self._check_params_valid(module) and self._check_fs_attributes(module)
 
+    @abc.abstractmethod
+    def _check_params_valid(self, module):
```

Comment:
```suggestion
    def _check_params_valid(self):
```
We pass the module object to the class' constructor, we could use ```self.module = module``` and reuse.
I think it does make sense to pass parameters to methods because it's much clear, but the ```module``` object looks a bit general.

Reply:
Passing `module` to some functions (but not all) is a common pattern for all `openssl_*` modules. I'd like to change that eventually, but I'd prefer not to do that in this PR :)

Reply:
yep, of course, would be cool, i only pointed that out :)